rasterio/_copy.pyx

```diff
@@ -1,48 +1,76 @@
 """Raster copying."""
 
+
 include "gdal.pxi"
 
 import logging
-import os
-import os.path
 
-from rasterio._err cimport exc_wrap_int, exc_wrap_pointer
+from rasterio._io cimport DatasetReaderBase
+from rasterio._err cimport exc_wrap_pointer
+from rasterio.env import ensure_env
 
 
 log = logging.getLogger(__name__)
 
 
-cdef class RasterCopier:
+@ensure_env
+def copy(src, dst, driver='GTiff', strict=False, **creation_options):
```

Comment:
I don't use it very often either, but if you need a copy of a raster this bypasses all of Rasterio's Numpy transactions.  On second look this might be useful, especially since the returned dataset handle may be in update mode depending on the driver.

```
with rasterio.copy('original.tif', 'copy.tif') as dst:
    dst.write(...)
```

 I'll poke around a bit more.  The only way to support both the above and just plain `rasterio.copy('original.tif', 'copy.tif')` would be with a class.